std-bits/tableau/src/main/java/org/enso/tableau/HyperFormat.java

```diff
@@ -330,14 +333,36 @@ public static Column[] readTable(
     }
   }
 
-  public static void writeTable(String path, String schemaName, String tableName, Table table)
+  public static String[] writeTable(
+      String path,
+      String schemaName,
+      String tableName,
+      Table table,
+      boolean append,
+      boolean matchColumnsByName,
+      boolean throwDontWarn)
       throws IOException {
+    List<String> warningUnmatchedColumns = new ArrayList<>();
     getProcess();
     try (var connection =
         new Connection(process.getEndpoint(), path, CreateMode.CREATE_IF_NOT_EXISTS)) {
-      var tableDef = createTable(schemaName, tableName, table.getColumns(), connection);
-      insertData(table, tableDef, connection);
+      TableDefinition tableDef;
+      if (append && tableExists(schemaName, tableName, connection)) {
+        tableDef = connection.getCatalog().getTableDefinition(new TableName(schemaName, tableName));
+      } else {
+        tableDef = createTable(schemaName, tableName, table.getColumns(), connection);
+      }
+      insertData(
+          table, tableDef, connection, matchColumnsByName, warningUnmatchedColumns, throwDontWarn);
+      connection.close();
     }
+    return warningUnmatchedColumns.toArray(String[]::new);
```

Comment:
feels odd to return `new string[0]` on success.
The other warning style is to pass a `ProblemAggregator` object in and read the issues from that and return a simple status here.